### PR TITLE
github actions: Add build and unit test checks

### PR DESCRIPTION
**Pull request by oshoval**
Signed-off-by: Or Shoval <oshoval@redhat.com>

### CODE REVIEW

File: .github/workflows/check.yaml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    - name: Set up Go
+      uses: actions/setup-go@v3
+      with:
+        go-version: 1.19
+
+    - name: Build
+      run: make build
+
+    - name: Linters
+      uses: golangci/golangci-lint-action@v3
```
Comment (AlonaKaplan) [inline .github/workflows/check.yaml:32]: It would be nice to add the linting to the Makefile and then invoke the `make lint` here. Also similar to nmpolicy.

File: .github/workflows/check.yaml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+jobs:
+  build-linters-unit-tests:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/setup-python@v2
+      with:
+        python-version: 3.x
+
+    - run: |
+        pip3 install --user --upgrade j2cli
```
Comment (AlonaKaplan) [inline .github/workflows/check.yaml:18]: why is this needed?

File: .github/workflows/check.yaml [hunk 3]
```diff
@@ -0,0 +1,38 @@
+name: Build, linters, and unit tests
+
+on:
+  push:
+    branches: [ "main" ]
+  pull_request:
+    branches: [ "main" ]
+
+jobs:
+  build-linters-unit-tests:
```
[the remark this answers is not in the document]
Comment (oshoval) [reply .github/workflows/check.yaml:10]: no need to checkout code few times, but anyhow will align to nmpolicy

File: .github/workflows/check.yaml (hunk 1 above)
Comment (oshoval) [reply .github/workflows/check.yaml]: will do it on a follow PR, this one adds the `make build` and `make test` (removed the previous lint one)

File: .github/workflows/check.yaml (hunk 2 above)
Comment (oshoval) [reply .github/workflows/check.yaml]: removed as part of rest of the changes (aligning it to NMPolicy)

**Comment by oshoval** [comment]
Addressed comments please see inline

File: .github/workflows/check.yaml [hunk 4]
```diff
@@ -0,0 +1,49 @@
+# Copyright 2022 The Kubevirt Authors.
```
Comment (oshoval) [inline .github/workflows/check.yaml:1]: Not related to this PR:

We need to change the global lic of this repo as well (and the gitignore)
both those files were created with some template but not updated yet

**Comment by oshoval** [comment]
Once we see it runs the `make test` we can remove the lane from prow (it is still on other open PR)

File: .github/workflows/check.yaml (hunk 4 above)
Comment (AlonaKaplan) [reply .github/workflows/check.yaml:1]: Please keep it `Copyright 2022.` like the other files have.
Once will will update the template I guess it will be `Copyright 2022 The KubeSecondaryDNS Authors.`

Comment (oshoval) [reply .github/workflows/check.yaml]: done

**Comment by oshoval** [comment]
addressed comments

File: .github/workflows/check.yaml (hunk 4 above)
Comment (oshoval) [reply .github/workflows/check.yaml]: Note that some files in this repo have
"# Copyright 2018-2022 Red Hat, Inc."
so maybe that we have all with the same license

and that imo the main LICNESE also need to have the who / year
